src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+                return RELATION::ERODED;
+            case gsoap_resqml2_0_1::resqml2__ContactMode__proportional:
+                return RELATION::CONFORMABLE;
+            default:
+                ringmesh_assert_not_reached;
+                return RELATION::CONFORMABLE;
+            }
+        }
+    } // namespace
+
+    gmge_id GeoModelBuilderRESQMLImpl::find_layer(
+        const Region3D& region ) const
+    {
+        std::set< const Interface3D* > horizons;
+        for( auto b : range( region.nb_boundaries() ) )
+        {
+            const Surface3D& surface = region.boundary( b );
+            const Interface3D& interface = static_cast< const Interface3D& >(
+                surface.parent( Interface3D::type_name_static() ) );
+            if( GMGE::is_stratigraphic_limit( interface.geological_feature() ) )
+            {
+                horizons.insert( &interface );
+            }
+        }
+
+        if( horizons.size() >= 1 && horizons.size() <= 2 )
+        {
+            // TODO the following detection is not working if the region doesn't
+            // contain a single face of one or even both horizons
+            std::vector< gmge_id > horizons_list;
+            for( auto h : horizons )
+            {
+                horizons_list.push_back( h->gmge() );
+            }
+            std::sort( horizons_list.begin(), horizons_list.end() );
+            for( auto& unit : unit_2_info_ )
+            {
+                std::vector< gmge_id > cur_horizons_list;
+                if( unit.second.interface_top_.is_defined() )
+                {
+                    cur_horizons_list.push_back( unit.second.interface_top_ );
+                }
+                if( unit.second.interface_base_.is_defined() )
+                {
+                    cur_horizons_list.push_back( unit.second.interface_base_ );
+                }
+                std::sort( cur_horizons_list.begin(), cur_horizons_list.end() );
+
+                if( cur_horizons_list.size() != horizons_list.size() )
+                {
+                    continue;
+                }
+                std::vector< gmge_id > result;
+                std::set_intersection( horizons_list.begin(),
+                    horizons_list.end(), cur_horizons_list.begin(),
+                    cur_horizons_list.end(), std::back_inserter( result ) );
+                if( result.size() == horizons_list.size() )
+                {
+                    auto id = interp_2_geo_entity_.find( unit.first );
+                    ringmesh_assert( id != interp_2_geo_entity_.end() );
+                    return id->second;
+                }
+            }
+        }
+        return gmge_id();
+    }
+
+    void GeoModelBuilderRESQMLImpl::read_strati_column( EpcDocument& pck )
+    {
+        const std::vector< RESQML2_0_1_NS::StratigraphicColumn* >
+            stratiColumnSet = pck.getStratigraphicColumnSet();
+        // TODO only one column for the moment
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1135)